packages/theme/src/cli/commands/theme/check.ts

```diff
@@ -133,7 +133,7 @@ export default class Check extends ThemeCommand {
       return
     }
 
-    const {offenses, theme} = await themeCheckRun(path, config)
+    const {offenses, theme} = await themeCheckRun(path, config, (message) => outputDebug(message))
```

Comment:
I have two choices here:
1. Use `outputDebug` (which is nice because it does all the colouring/timestamp/--verbose/SHOPIFY_FLAG_VERBOSE good stuff)
2. Use `console.error` directly

The problem with (1) is that, because of #3863, I can't really use it unless I change the default logger of outputDebug to be `consoleError`. I wouldn't be able to use it because Shopify/theme-check-action would capture STDOUT of `shopify theme check --output json --verbose` and expect JSON. But since outputDebug dumps to STDOUT, we'd have malformed JSON to deal with. 

(2) is a more local fix if you want to make the default change in a different PR.